version: '2'
services:
   webapp:
      image: "heliumdatastage/jupyter-datascience:v0.0.3"
      volumes:
        - ${DATA_VOLUME}:/data
        - ${USER_VOLUME}:/home/jovyan/pd
      entrypoint:
        - /bin/bash
        - -c
        - start.sh jupyter lab --LabApp.token="" --NotebookApp.base_url="/lw-workspace/proxy"
      environment:
        - NAMESPACE=helium
        - HOSTNAME=helium.planx-pla.net
        - FRAME_ANCESTORS=https://helium.planx-pla.net
      ports: 
        - "${SERVICE_PORT}:8888"
      deploy:
        resources:
          limits:
            cpus: '2'
            memory: 2048M
          reservations:
            cpus: '2'
            memory: 2048M
   nginx:
      image: "quay.io/cdis/ws-nginx-proxy:2021.01"
      ports:
        - "${SERVICE_PORT}:7780"
      environment:
        - "APP_PORT=8888"
      healthcheck:
        test: ["CMD", "curl", "-f", "http://localhost:7780/_proxy_status"]
        interval: 1m30s
        timeout: 10s
        retries: 3
